module-system/sov-modules-macros/src/common.rs

```diff
@@ -123,9 +123,16 @@ impl<'a> StructDef<'a> {
     }
 }
 
-/// Gets type parameter from `Generics` declaration.
-pub(crate) fn parse_generic_params(generics: &Generics) -> Result<Ident, syn::Error> {
-    let generic_param = match generics.params.first().unwrap() {
+/// Gets the type parameter's identifier from [`syn::Generics`].
+pub(crate) fn get_generics_type_param(
+    generics: &Generics,
+    error_span: Span,
+) -> Result<Ident, syn::Error> {
+    let generic_param = match generics
+        .params
+        .first()
+        .ok_or_else(|| syn::Error::new(error_span, "No generic parameters found"))?
```

Comment:
(Out-of-scope) I know this logic wasn't introduced in this PR, but do we document anywhere that the first generic must always be the `Context` type?

Reply:
There seems to be some relevant docs here: https://github.com/Sovereign-Labs/sovereign-sdk/blob/04a37bff3a172c5e4c13d07738c532178c3f5c76/module-system/README.md#L12-L36

Reply:
I might be missing something, but I don't see anything about the _ordering_ constraint on the generics (i.e. that `C` must come first). Maybe we can add that in here?

Reply:
I clarified a bit in https://github.com/Sovereign-Labs/sovereign-sdk/pull/517/commits/12035eab403340f32ad61d9b904b3dee611a492f :+1: